https://issues.redhat.com/browse/ACM-19059--known issue for SiteConfig (#7656)
open-cluster-management/rhacm-docs: release_notes/known_issues_cluster.adoc

Or consider a troubleshooting topic.
////

Review the known issues for _Cluster management_ with {acm-short}. The following list contains known issues and limitations for this release, or known issues that continued from the previous release.

For Cluster management with the stand-alone {mce-short} known issues and limitations, see link:../clusters/release_notes/mce_known_issues.adoc#known-issues-mce[Cluster lifecycle known issues and limitations] in the {mce-short} documentation.

[#cluster-local-offline-reimport]
== SiteConfig Operator fails to start with {acm-short} 2.13
//2.13:19059

When you enable the SiteConfig Operator for {acm-short} 2.13, you get the following error:

+
[source,bash]
----

> $ oc logs siteconfig-controller-manager-5874db7bb8-vhqmj kube-rbac-proxy
> unknown flag: --http2-disable
> Usage of /usr/local/bin/kube-rbac-proxy:
> unknown flag: --http2-disable

Reviewer: ```suggestion
$ oc get pods -A | grep siteconfig
rhacm                                              siteconfig-controller-manager-xxx-xxx                                               1/2     CrashLoopBackOff   39 (4m5s ago)   179m
```

----

The SiteConfig Operator references the `--http2-disable parameter` in the upstream `kube-rbac-proxy` image within the deployment manifest. However, the {acm-short} `kube-rbac-proxy` image does not support this parameter.

To work around this known issue, see the following procedure:

. Pause the `MultiClusterHub` custom resource. 

> . Manually update the deployment manifest. Remove the `--http2-disable parameter` from the `kube-rbac-proxy` container. Run the following command:

Reviewer: ```suggestion
. Manually update the `siteconfig-controller-manager` deployment manifest. Remove the `--http2-disable parameter` from the `kube-rbac-proxy` container. Run the following command:
```

+
[source,bash]
----
oc edit deploy siteconfig-controller-manager -n rhacm 
----

> . After you apply this change, restart the affected pods to ensure your update is successful.

Reviewer: ```suggestion
. After you apply this change, verify that the affected pods are restarted to ensure your update is successful.
```

Or consider a troubleshooting topic.
////

Review the known issues for _Cluster management_ with {acm-short}. The following list contains known issues and limitations for this release, or known issues that continued from the previous release.

For Cluster management with the stand-alone {mce-short} known issues and limitations, see link:../clusters/release_notes/mce_known_issues.adoc#known-issues-mce[Cluster lifecycle known issues and limitations] in the {mce-short} documentation.

[#cluster-local-offline-reimport]
== SiteConfig Operator fails to start with {acm-short} 2.13
//2.13:19059

When you enable the SiteConfig Operator for {acm-short} 2.13, you get the following error:

[source,bash]
----
$ oc get pods -A | grep siteconfig
rhacm                                              siteconfig-controller-manager-xxx-xxx                                               1/2     CrashLoopBackOff   39 (4m5s ago)   179m
----

> The SiteConfig Operator references the `--http2-disable parameter` in the upstream `kube-rbac-proxy` image within the deployment manifest. However, the {acm-short} `kube-rbac-proxy` image does not support this parameter.

Reviewer 2: ```suggestion
The SiteConfig Operator references the `--http2-disable` parameter in the upstream `kube-rbac-proxy` image within the deployment manifest. However, the {acm-short} `kube-rbac-proxy` image does not support this parameter.
```

> To work around this known issue, see the following procedure:

Reviewer 2: ```suggestion
To work around this known issue, complete the following procedure:
```

To work around this known issue, see the following procedure:

. Run the following command to pause the `MultiClusterHub` custom resource:

+
----
oc annotate mch -n rhacm multiclusterhub mch-pause=true --overwrite=true
----

> . Manually update the `siteconfig-controller-manager` deployment manifest. Remove the `--http2-disable parameter` from the `kube-rbac-proxy` container. Run the following command:

Reviewer: ```suggestion
. Manually update the `siteconfig-controller-manager` deployment manifest. Remove the `--http2-disable parameter` from the `kube-rbac-proxy` container. Run the following command for the manual update:
```

> . Manually update the `siteconfig-controller-manager` deployment manifest. Remove the `--http2-disable parameter` from the `kube-rbac-proxy` container. Run the following command for the manual update:

Reviewer 2: ```suggestion
. Manually update the `siteconfig-controller-manager` deployment manifest. Remove the `--http2-disable` parameter from the `kube-rbac-proxy` container. Run the following command for the manual update:
```

Reviewer: Nice catch @amolnar-rh

> . Manually update the `siteconfig-controller-manager` deployment manifest. Remove the `--http2-disable parameter` from the `kube-rbac-proxy` container. Run the following command for the manual update:

Reviewer 2: This line seems to have 2 different steps:
1) Update the deployment manifest
2) Remove the parameter

By running the command, does it do both in one step?

Reviewer: The parameter needs to be manually removed by editing the deployment manifest with the given command.

+
[source,bash]
----
oc edit deploy siteconfig-controller-manager -n rhacm 
----

. After you apply this change, verify that the affected pods are restarted to ensure your update is successful.

> 

Reviewer 3: do we need to unpause MCH?

> 

Reviewer: No, unpausing will result in the changes being reverted.
